lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -50,41 +98,84 @@ extension DomWindowExtension on DomWindow {
     );
   }
 
+  @JS('fetch')
+  external JSPromise _fetch1(JSString url);
+  @JS('fetch')
+  external JSPromise _fetch2(JSString url, JSAny headers);
+
   // ignore: non_constant_identifier_names
   external DomURL get URL;
-  external bool dispatchEvent(DomEvent event);
-  external DomMediaQueryList matchMedia(String? query);
-  DomCSSStyleDeclaration getComputedStyle(DomElement elt,
-          [String? pseudoElt]) =>
-      js_util.callMethod(this, 'getComputedStyle', <Object>[
-        elt,
-        if (pseudoElt != null) pseudoElt
-      ]) as DomCSSStyleDeclaration;
+
+  @JS('dispatchEvent')
+  external JSBoolean _dispatchEvent(DomEvent event);
+  bool dispatchEvent(DomEvent event) => _dispatchEvent(event).toDart;
+
+  @JS('matchMedia')
+  external DomMediaQueryList _matchMedia(JSString? query);
+  DomMediaQueryList matchMedia(String? query) => _matchMedia(query?.toJS);
+
+  @JS('getComputedStyle')
+  external DomCSSStyleDeclaration _getComputedStyle1(DomElement elt);
+  @JS('getComputedStyle')
+  external DomCSSStyleDeclaration _getComputedStyle2(
+      DomElement elt, JSString pseudoElt);
+  DomCSSStyleDeclaration getComputedStyle(DomElement elt, [String? pseudoElt]) {
+    if (pseudoElt == null) {
+      return _getComputedStyle1(elt);
+    } else {
+      return _getComputedStyle2(elt, pseudoElt.toJS);
+    }
+  }
+
   external DomScreen? get screen;
-  external double requestAnimationFrame(DomRequestAnimationFrameCallback callback);
+
+  @JS('requestAnimationFrame')
+  external JSNumber _requestAnimationFrame(JSFunction callback);
+  double requestAnimationFrame(DomRequestAnimationFrameCallback callback) =>
+      _requestAnimationFrame(callback.toJS).toDart;
+
+  @JS('postMessage')
+  external JSVoid _postMessage1(JSAny message, JSString targetOrigin);
```

Comment:
Honestly, it was kind of the 'Nth' variation. Do you have a preference that it be the number of arguments? I do not have strong feelings, and I'm happy to change it.

Reply:
The first time I looked at syntax like this was when people were poking fun at Scala's [`Function22` Trait](https://www.scala-lang.org/api/2.12.9/scala/Function22.html) (a Function that accepts 22 parameters), and I think I mapped the X in `FunctionX` to "X number of parameters" rather than the Nth variation. I don't have strong feelings at all (as long as we're consistent). Also if we ever wanted to "fix" this, it'd be a fully internal rename, so I don't think it matters that much :P

Reply:
TL;DR: Leave this as is, until somebody gets super offended and does the find-and-replace. It might even be you in a few months! :P